### ISSUE TITLE
The treatmentDomain branch is functioning in Private Browsing mode

### ISSUE THREAD

**Issue by SoftVision-CosminMuntean** [mozilla/fingerprinting-blocking-methods-study-addon#11]
[Notes]:
- Based on the [TESTPLAN.md](https://github.com/mozilla/fingerprinting-blocking-methods-study-addon/blob/master/TESTPLAN.md) from GitHub repository all branches should not be functioning/interacting with private browsing mode. 
- Probably this issue is reproducible because of the issue logged in #10.
- The issue is not reproducible on the treatmentScript branch. 

[Affected Platforms]:
- All Windows
- All Mac
- All Linux

[Affected Versions]:
- Firefox 69.0 Unbranded build
- Firefox 70.0.1 Unbranded build
- Firefox 71 Unbranded build

[Prerequisites]:
- Have a Firefox 69, 70 or 71 Unbranded build. 
- Have access to the stage delivery console.
- Have an active branched-addon recipe for the Fingerprinting Implementation Retention Study V1. You can use my [created recipe](https://delivery-console.stage.mozaws.net/recipe/882).
- Have a user.js file that contains all the prefs needed to be targeted by the Normandy recipe. You can use my created [user.js](https://drive.google.com/open?id=12j8J7WTMU-ZvHjcUhcutB09Qau_GQ24_).

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites.
2. In the profile directory copy the user.js file.
3. Restart the browser.
4. After enrolling in the treatmentDomain open a Private Window (Ctrl/Cmd + Shift + P).
5. Navigate to a website that contain Fingerprinting tracks eg: https://www.ssense.com/en-us/account/login.
6. Click the "Shield" icon from the URL bar.
7. Observe the "Content Blocking" section from the panel.

[Expected result]:
- Fingerprinters are detected but not blocked.

[Actual result]:
- Fingerprinters are detected and blocked.

**Comment by birdsarah** [comment]
Hi, I apologize for my test plan not being clear. In private browsing mode the expected result, for all branches, is `Fingerprinters are detected and blocked.` 

Fingerprinting blocking is, by default, in release, turned on in Private Browsing Mode. This experiment should not affect or change that. That is, this experiment should not reduce, in any branch, the ETP fingerprinting protections on by default for users. In fact, my study addon should not interact with PBM at all.

cc @motin can you double check that I set this up correctly. Here's the link to my three manifests:
* https://github.com/mozilla/fingerprinting-blocking-methods-study-addon/blob/master/src/treatmentDomain/manifest.json.tmpl
* https://github.com/mozilla/fingerprinting-blocking-methods-study-addon/blob/master/src/treatmentScript/manifest.json.tmpl
* https://github.com/mozilla/fingerprinting-blocking-methods-study-addon/blob/master/src/control/manifest.json.tmpl

---

@SoftVision-CosminMuntean I am more concerned that you found that the "issue" is not reproducible on the treatmentScript branch - because that would be a bug. Can you post a screencast?

**Comment by SoftVision-CosminMuntean** [comment]
@birdsarah I have retested this issue when enrolled in the treatmentScript branch and it seems that in Private Window the Fingerprinters are not detected, but they are detected in normal Windows.
Here is a screen recording with the issue: [link](https://drive.google.com/file/d/1ujrluhnuwnjawqgyM4mqfR03TCAi1TNO/view?usp=sharing).

However, it seems that the resources are blocked under the "Tracking Content" in Private Window, so probably this is expected.
It is a bit confusing because I expected to see the message that fingerprinters are detected and blocked in the "Content Blocking" doorhanger. Similar behavior can be observed also in a normal window. I have logged a separate issue for this in #12.

**Comment by birdsarah** [comment]
Thank you for your continued diligence, it's much appreciated.

Everything is working as expected. This, unfortunately, includes a bug in the display of ETP in private browsing mode, which is fixed in 72.

To recap without study addon, when visiting skyscanner.com: 

* Behavior in ff72:
 * In normal browsing with default settings: perimetrix.net is not blocked and is listed as a fingerprinter that hasn't been blocked in door hanger
 * In private browsing with default settigns: perimetrix.net is blocked but is listed as a fingerprinter
* Behavior in ff70 and ff71:
 * In normal browsing with default settings: perimetrix.net is not blocked and is listed as a fingerprinter that hasn't been blocked in door hanger
 * In private browsing with default settigns: perimetrix.net is blocked but is not listed as a fingerprinter, but instead as "tracking content"
* Behavior in ff69:
 * Same blocking effects as above, but fingerprinters are not a category in door hanger for us to observe

The behavior in ff70 and ff71 is arguably a UX bug. It has been improved in ff72. I believe the confusion comes down to the fact that the fingerprinters are always a subset of trackers and therefore can be classified as either.

All of the above is the default behavior that we expect to see in the treatmentScript branch. However, in the treatmentScript branch we also expect to see, in the browser console, in normal browsing only, a note that a specific script from perimetrix.net was blocked. Other content from perimetrix.net is allowed.

I will agree this is a confusing interaction with the Firefox UI. The door hanger says the fingerprinting domain perimetrix.net was allowed, but we have, behind the scenes blocked a specific fingerprinting script from this domain.

This lack of fine grained UI is a known limitation of the study, and we believe it's acceptable. 

ping @mlopatka - we have already discussed this, but as @SoftVision-CosminMuntean brought it up, just wanted to check that you are still on board with this approach. If yes, then we can close this issue.

(I am following up with anti-tracking team to verify that my understanding of UI is correct. If it is not, I will reopen this issue).

**Comment by SoftVision-CosminMuntean** [comment]
@birdsarah thank you for all the provided details and for the clarification, I appreciate it.
We have updated our created test cases accordantly and we will start running them as soon as the builds are signed.

**Comment by birdsarah** [comment]
I followed up with @englehardt and my understanding was not correct. However, it does not affect the resolution of this issue. I will recap everything so we're clear on what's happening.

In private browsing mode, by default, fingerprinting protection is not toggled on. There is no specific feature flag to turn on fingerprinting protections in private browsing mode. Although it is not explicitly on as a feature, it is on functionally. In private browsing mode, by default tracking content is blocked. Because all fingerprinting domains are a subset of the tracking domains (at least for the time being) all fingerprinting content will be blocked in private browsing mode by the tracking content setting. This is the reason why we see the blocking under the tracking category in private browsing mode.

This experiment will not change that, we are only adding fingerprinting protections in standard browsing mode. So the expectations are as follows:
* control branch - fingerprinters are allowed in standard browsing, fingerprinters are blocked in private browsing mode and listed as tracking content
* treatmentDomain branch - fingerprinters are blocked in standard browsing, fingerprinters are blocked in private browsing mode and listed as tracking content
* treatmentScript branch - fingerprinters will show as allowed in the ETP UI but individual fingerprinting scripts will be blocked and this is logged in the browser console, fingerprinters are blocked in private browsing mode and listed as tracking content

**Comment by SoftVision-CosminMuntean** [comment]
@birdsarah thank you for the provided details are really useful.

However, while testing the treatmentDomain branch, I have encountered a different behavior in Private Window that you described above.

> treatmentDomain branch - fingerprinters are blocked in standard browsing, fingerprinters are blocked in private browsing mode and listed as tracking content

When you are enrolled in the treatmentDomain branch the "privacy.trackingprotection.fingerprinting.enabled" pref is set to "true". Because of this, in Private Window the fingerprinters are detected and blocked under "Fingerprinters" and not under the "tracking content".
Here is a screen recording with the described behavior: [link](https://drive.google.com/file/d/1trvzk5dGmf5mVmq07Ra6uGo5YD1eR2dT/view?usp=sharing). 

I'm a bit confused and not sure if this is intended or we have a new bug.

**Comment by birdsarah** [comment]
@SoftVision-CosminMuntean. Thank you again for being so detailed.

I can also reproduce your finding. 

TLDR: This was not my intention. But it is acceptable for the experiment. Moreover it is not fixable.

Full explanation. 
1) Acceptability - The purpose of the test case was to ensure that protections were not reduced in private browsing mode. This reduction could potentially only happen in control and treatmentScript, where fingerprinting is explicitly off. However, given what we've learned in this thread, that original fear was unwarranted as protection would have existed through the tracking content protection. In the case of treatmentDomain, the fingerprinting is on, there is no loss of protection, so the behavior you're observing is acceptable.
2) Implementation not fixable - The addon is setup to not interact with private browsing mode. (see `incognito: not_allowed` in manifest.json in all adons). This is working, as you've tested, with treatmentScript. I believe it's not working here because a prefFlip at startup is very different. And that pref flip remains in force in private browsing. This is fine in this case, but this is a very useful learning for me.

I have updated the TESTPLAN accordingly: https://github.com/mozilla/fingerprinting-blocking-methods-study-addon/commit/0be058a7d676433d96bc4258352ae52ee4a7c47b
